### PR TITLE
add `prepare_isf_2023_scenario()`

### PR DESCRIPTION
**Pull request by cjyetman**
- work towards https://github.com/RMI-PACTA/workflow.scenario.preparation/issues/9
- precedes https://github.com/RMI-PACTA/workflow.scenario.preparation/pull/24

The `iso_2023` object prepared here has some differences with the `pacta.scenario.preparation::iso_2023` object, but I have confirmed with @Antoine-Lalechere that they are probably desirable:
- some geography names in the old version have been fixed to what appears to be "standard" in other scenario data objects ("Saudi_Arabia" -> "Saudi Arabia", "South_Africa" -> "South Africa", "South_Korea" -> "South Korea", "USA" -> "US")
- Coal and Oil&Gas values in the old version appear to unintentionally exclude "- Hard coal & lignite" and "- Oil products" values. I believe this is caused by an inner_join with the technology_bridge that does not include those technologies (see [here](https://github.com/RMI-PACTA/pacta.scenario.preparation/blob/main/data-raw/technology_bridge.csv) and [here](https://github.com/RMI-PACTA/pacta.scenario.preparation/blob/1523d2ffb4afcddd4f96b8b68030971998967fde/data-raw/prepare_isf2023.R#L159-L194))

the following code finds no differences
``` r
waldo::compare(
  dplyr::filter(
    dplyr::arrange(pacta.scenario.preparation::isf_2023, scenario, scenario_geography, sector, technology),
    !sector %in% c("Oil&Gas", "Coal")
  ),
  dplyr::filter(
    dplyr::mutate(
      dplyr::arrange(isf_2023, scenario, scenario_geography, sector, technology),
      scenario_geography = dplyr::case_when(
        scenario_geography == "Saudi Arabia" ~ "Saudi_Arabia",
        scenario_geography == "South Africa" ~ "South_Africa",
        scenario_geography == "South Korea" ~ "South_Korea",
        scenario_geography == "US" ~ "USA",
        .default = scenario_geography
      )
    ),
    !sector %in% c("Oil&Gas", "Coal")
  ),
  tolerance = 1e-15
)
```

### CODE REVIEW

**Review by Antoine-Lalechere — APPROVED** [review]
LGTM

File: DESCRIPTION [hunk 1]
```diff
@@ -13,17 +13,20 @@ Suggests:
...
 Imports: 
     dplyr,
     glue,
     janitor,
     pacta.data.validation,
+    purrr,
     rlang,
     stringr,
     tidyr,
+    unpivotr,
```
Comment (jdhoffa) [inline DESCRIPTION:29]: NB: What is this package?

File: R/prepare_isf_2023_scenario.R [hunk 2]
```diff
@@ -0,0 +1,306 @@
...
+    dplyr::filter(col == 17)
+
+  partitions <- unpivotr::partition(x, corners)
+  tables <- dplyr::pull(partitions, "cells")
+
+  tables[[1]] %>%
+    dplyr::filter(!.data[["is_blank"]]) %>%
+    unpivotr::behead("up-left", "table") %>%
+    unpivotr::behead("up", "year", formatters = list(numeric = as.double)) %>%
+    unpivotr::behead("left", "technology") %>%
```
Comment (jdhoffa) [inline R/prepare_isf_2023_scenario.R:97]: NB: lol yeah this whole `tidyxl` + `unpivotr` thing feels a bit over-engineered. But still cool that it exists!

**Review by jdhoffa — APPROVED** [review]
My review was mainly:
* syntax
* check that `waldo` result is valid

but LGTM :-)

File: DESCRIPTION (hunk 1 above)
Comment (cjyetman) [reply DESCRIPTION:29]: this is the counterpart of {tidyxl}... {tidyxl} does the importing from XLSX files, {unpivotr} does the processing/reshpaing/etc. of tidyxl-style data frames

Comment (jdhoffa) [reply DESCRIPTION:29]: yup makes sense, i put that together after reviewing the later parts haha

File: R/prepare_isf_2023_scenario.R (hunk 2 above)
Comment (cjyetman) [reply R/prepare_isf_2023_scenario.R:97]: power vs. less-complexity ⚖️ 
🤣 

https://nacnudus.github.io/spreadsheet-munging-strategies/
> Tidyxl and unpivotr are much more complicated than readxl, and that’s the point. Tidyxl and unpivotr give you more power and complexity when you need it.

https://nacnudus.github.io/tidyxl/index.html#similar-projects
> [tidyxl](https://github.com/nacnudus/tidyxl) was originally derived from [readxl](https://github.com/tidyverse/readxl) and still contains some of the same code, hence it inherits the GPL-3 licence. [readxl](https://github.com/tidyverse/readxl) is intended for importing tabular data with a single row of column headers, whereas [tidyxl](https://github.com/nacnudus/tidyxl) is more general, and less magic.

**Pull request merged**